src/alda/sound/midi.clj

```diff
@@ -199,3 +216,32 @@
       .getChannels
       (pmap stop-channel!)
       doall)))
+
+(defn play-sequence!
+  "Plays a sequence on a java midi sequencer.
+
+  Execute callback when sequence is done."
+  [sequence promise!]
```

Comment:
I suggest renaming `promise!` to `done-fn`, so it's extra clear that it is a function to be called when playback is done.